electrum/gui/qt/wizard/wallet.py

```diff
@@ -1250,25 +1249,35 @@ def apply(self):
 
 
 class WCWalletPasswordHardware(WalletWizardComponent):
+
     def __init__(self, parent, wizard):
         WalletWizardComponent.__init__(self, parent, wizard, title=_('Encrypt using hardware'))
         self.plugins = wizard.plugins
-
-        self.playout = PasswordLayoutForHW(MSG_HW_STORAGE_ENCRYPTION)
+        # TODO: PasswordLayout assumes a button, refactor PasswordLayout
+        # for now, fake next_button.setEnabled
+        class Hack:
+            def setEnabled(self2, b):
+                self.valid = b
+        self.next_button = Hack()
+        self.playout = PasswordLayoutForHW(
+            MSG_HW_STORAGE_ENCRYPTION,
+            kind=PW_NEW,
+            OK_button=self.next_button,
+        )
         self.layout().addLayout(self.playout.layout())
         self.layout().addStretch(1)
 
         self._valid = True
 
     def apply(self):
-        self.wizard_data['encrypt'] = self.playout.should_encrypt_storage()
+        self.wizard_data['encrypt'] = self.playout.should_encrypt_storage_with_xpub()
         _name, _info = self.wizard_data['hardware_device']
         device_id = _info.device.id_
         client = self.plugins.device_manager.client_by_id(device_id, scan_now=False)
         # client.handler = self.plugin.create_handler(self.wizard)
         # FIXME client can be None if it was recently disconnected.
         #       also, even if not None, this might raise (e.g. if it disconnected *just now*):
-        self.wizard_data['password'] = client.get_password_for_storage_encryption()
+        self.wizard_data['xpub_password'] = client.get_password_for_storage_encryption()
```

Comment:
this might spread far, and the wizard_data dict is fully opaque to static analysers :(
Anyway:
```patch
diff --git a/electrum/gui/qt/__init__.py b/electrum/gui/qt/__init__.py
index 33cc3af744..db95cc5a68 100644
--- a/electrum/gui/qt/__init__.py
+++ b/electrum/gui/qt/__init__.py
@@ -443,7 +443,7 @@ class ElectrumGui(BaseElectrumGui, Logger):
         else:
             wallet_file = d['wallet_name']
 
-        password = d.get('password') or None  # convert '' to None
+        password = d.get('password') or d.get('xpub_password') or None  # convert '' to None
 
         try:
             wallet = self.daemon.load_wallet(wallet_file, password, upgrade=True)

```